File: python/courtlistener/api/aba_ratings.py
"""
ABA Ratings API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class ABARatingsAPI(BaseAPI):
    """API for accessing ABA ratings data."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "aba-ratings/"
    
    def list(
        self,
        person: Optional[int] = None,
        **kwargs
    ) -> Dict[str, Any]:
        """
        List ABA ratings.
        
        Args:
            person: Filter by person ID
            **kwargs: Additional query parameters
        
        Returns:
            API response with paginated list of ABA ratings
        """
        params = {}
        if person is not None:
            params['person'] = person
        
        params.update(kwargs)
        return self.client.get(self.endpoint, params=params)
    
    def get(self, rating_id: int) -> Dict[str, Any]:
        """
        Get a specific ABA rating by ID.
        
        Args:
            rating_id: ABA rating ID
        
        Returns:
            ABA rating data
        """
        return self.client.get(f"{self.endpoint}{rating_id}/")
    
    def paginate(
        self,
        person: Optional[int] = None,
        **kwargs
    ):
        """
        Get paginated ABA ratings.
        
        Args:
            person: Filter by person ID
            **kwargs: Additional query parameters
        
        Returns:
            PageIterator for ABA ratings
        """
        params = {}
        if person is not None:
            params['person'] = person
        
        params.update(kwargs)
        return self.client.paginate(self.endpoint, params=params)

File: php/src/CourtListener/Api/AbaRatings.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * AbaRatings API endpoint
 */
class AbaRatings extends BaseApi
{
    protected string $endpoint = 'aba-ratings/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List aba ratings with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listAbaRatings(array $params = []): array
    {
        return $this->list($params);
    }

    /**
     * Get a specific aba ratings by ID
     *
     * @param int|string $id AbaRatings ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getAbaRatings($id, array $params = []): array
    {
        return $this->get($id, $params);
    }

    /**
     * Search aba ratings
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchAbaRatings(array $params = []): array
    {
        return $this->search($params);
    }
}
